mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +126,36 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    kwargs = {'decode_times': False,
+              'concat_dim': 'Time'}
+
+    autocloseFileLimitFraction = config.getfloat('input',
+                                                 'autocloseFileLimitFraction')
+
+    # get the number of files that can be open at the same time.  We want the
+    # "soft" limit because we'll get a crash if we exceed it.
+    softLimit = resource.getrlimit(resource.RLIMIT_NOFILE)[0]
+
+    # use autoclose if we will use more than half the soft limit on open files
```

Comment:
@xylar, can you please change the comment here? You haven't hard coded a factor of 2 anymore following my suggestion.